### PR TITLE
[mqtt.homeassistant] Add missing climate properties

### PR DESCRIPTION
**Pull request by VasekCejka**
Added support for humidity and preset_modes - two major attributes of HomeAssistant Climate specification not yet supported.  
`preset_mode` is useful for using different predefined scenarios. My motivation came from thermostatic radiator valves (TRV). There you can set directly on the valve the week schedule of desired temperatures. Because valves as passive devices tend to drop off the zigbee network and must be sometimes repaired (see other threads), it is very useful to have the schedule controlled directly by the valve even in offline state.
Other scenarios are e.g. workday / weekend / holiday... for TRV or HVAC devices.
Up to now only temperature settings was supported. On HVAC devices, the humidity control is also useful and important.

The new properties were added exactly in the template used by already supported properties.

Humidity channel should probably use PercentageValue, not NumberValue, but I'm very new to OH and not skilled in java, I'm not sure, how to use it properly. I ask someone more experienced to enhance it.

The OpenHAB user must perform re-discovery of already added Things to see new channels.

Humidity channels were not tested on real devices, because I have none.

### CODE REVIEW

**Comment by VasekCejka** [comment]
Thanks for leading me, hopefully done and next time will be better...

File: bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/Climate.java
Comment (lsiepel) [inline bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/Climate.java:326]: Regarding this comment, did you test it? To me this seems valid.

Comment (VasekCejka) [reply bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/Climate.java:326]: I have tested it with NumberValue and it works. I was looking at PercentageValue for a while, but I was afraid not to do some bigger bug. Probably you expirieced coders will be able to change it, if you will, without much troubles.

File: bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/Climate.java
Comment (ccutrer) [inline bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/Climate.java:56]: since there has never been a currentHumidity channel, we don't need to maintain it as deprecated. just remove this line (and change the conditional below that uses it to only use `CURRENT_HUMIDITY_CH_ID`)

File: bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/Climate.java
Comment (ccutrer) [inline bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/Climate.java:176]: this line is unnecessary; you only need to add `@SerializedName` when the name in the JSON doesn't exactly match the name of the field in the Java class

File: bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/Climate.java
Comment (ccutrer) [inline bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/Climate.java:188]: the default is empty list, not this. so `List.of()`

File: bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/Climate.java
Comment (ccutrer) [inline bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/Climate.java:244]: this should default to 99 (and don't make it `@Nullable`)

**Review by ccutrer — CHANGES_REQUESTED** [review]
can you also add a test that has the humidity channels?

File: bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/Climate.java
Comment (lsiepel) [reply bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/Climate.java:244]: Not 100, dew point?

File: bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/Climate.java
Comment (VasekCejka) [reply bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/Climate.java:358]: This change will cause failure of tests(`testTS0601ClimateNotSendIfOff`, `testClimate`), specifically one channel disappers - I suppose the one with power_topic. I'm not author of this test and don't have device with this property, and do not understand this behaviour. For this reason I suggest to pass this one to someone, who knows...

File: bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/Climate.java
Comment (VasekCejka) [reply bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/Climate.java]: thanks for all points, good school for me...
Done

File: bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/Climate.java
Comment (ccutrer) [inline bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/Climate.java:282]: you can remove this comment

Comment (VasekCejka) [reply bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/Climate.java]: sorry, done...

**Review by ccutrer — APPROVED** [review]
LGTM

**Comment by lsiepel** [comment]
Conflict needs to be resolved before this can be merged

**Comment by VasekCejka** [comment]
I have merged it, but I'm absolutly lost in what was happening and why new review requests arose... Sorry if I have something mistaken...

**Comment by ccutrer** [comment]
It's because you did a merge, not a rebase. See https://community.openhab.org/t/rebase-your-code-or-how-to-fix-your-git-history-before-requesting-a-pull/129358 for details on what happened.

**Comment by VasekCejka** [comment]
After couple of hours with lots of blood spilled I hope I have rebased the commit. But I was lost almost all the time, so I'm not sure. Please have a look at it and in case of need let me know...

**Comment by ccutrer** [comment]
You still have conflicts, so it must not be fully rebased. Did you `git fetch origin` (or `upstream` or whatever you named the openhab fork of the repo in your local repository) first?

**Comment by VasekCejka** [comment]
Sorry, I dont know. I'm running in circles. I'm not able to go throuht it, probably I miss some important point. On local PC git shows no conflict, no problem with spotless / build, and exactly the same code as here on web.
I am able to make fresh new fast pull request with final code and skip this pain - would it be OK?

**Comment by VasekCejka** [comment]
> I think you tried to update your local feature branch straight from openHAB main. This usually causes issues.
Yes, you are right, my mistake.
> 1. update your GitHub main. From openHAB main (23commits behind I see)
done
> 2. Merge your main branch into your  feature branch.
There is nothing to merge, because it was already merged with openHAB main. I tryied it even that (merge my main in local repository to my feature branch in local repository) - there was no single change on any file.
> 3. While doing 2 your IDE will show a merge conflict that you need to solve. Usually a text editor presents some kind of choice to accept incoming, current or combination.
I cannot see (neither yesterday) any merge conflict. Neithet pull request page does not speak about any merge conflict. I have now "Able to merge" message.
> 4. Commit the changes after the conflict is resolved, this merge commit will fix it all
The pull request is blocked by fail in automatic build process - the error is about need of spotless:apply. The trouble was that my local copy performed successful build. I have edited file directly here on github web and commited. It helped.
Thanx for patience

**Review by lsiepel — APPROVED** [review]
Thank you for this contribution. Now it is merged and it will be available in the next snapshot.
LGTM

**Pull request merged**
